processing/src/main/java/io/druid/query/extraction/ExtractionFn.java

```diff
@@ -68,11 +70,26 @@
 
   /**
    * Offers information on whether the extraction will preserve the original ordering of the values.
-   *
+   * <p/>
    * Some optimizations of queries is possible if ordering is preserved.  Null values *do* count towards
    * ordering.
    *
    * @return true if ordering is preserved, false otherwise
    */
   public boolean preservesOrdering();
+
+  /**
+   * A dim extraction can be of one of two types, renaming or rebucketing. In the `RENAME` case, a unique values is
+   * modified into another unique value. In the `REBUCKET` case, there is no longer a 1:1 relation between old dimension
+   * value and new dimension value
+   *
+   * @return {@link io.druid.query.extraction.ExtractionFn.ExtractionType} declaring what kind of manipulation this
+   * function does
+   */
+  public ExtractionType getExtractionType();
+
+  public static enum ExtractionType
+  {
+    REBUCKET, RENAME
```

Comment:
It seems we always have to explain what the difference between those two things mean. Maybe just calling them what they are, i.e. one-to-one and many-to-one would be more self-explanatory?